### PR TITLE
MM-34549 shared channels; add users to channel that were already sync'd

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR fixes a bug and adds a feature for shared channels:

The Bug: when creating new shared channels, users that had already been sync'd via another channel were not added to the new channel's member list, since the users were not sync'd again.  This PR sync's users per channel.

The Feature:  support custom statuses

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34549   bug
https://mattermost.atlassian.net/browse/MM-34682   feature

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: services/sharedchannel/sync_recv.go [hunk 1]
```diff
@@ -306,6 +317,8 @@ func (scs *Service) updateSyncUser(patch *model.UserPatch, user *model.User, cha
 				)
 			}
 		} else {
+			scs.server.GetStore().User().InvalidateProfileCacheForUser(update.New.Id)
```
Comment (agnivade) [inline services/sharedchannel/sync_recv.go:320]: This will not invalidate caches all across the cluster. Should we use `a.InvalidateCacheForUser` instead?

Comment (wiggin77) [reply services/sharedchannel/sync_recv.go]: Good point!   Fixed via https://github.com/mattermost/mattermost-server/pull/17361/commits/33adb5730da49e0f48f24a4d67a9bf54d9cb435b

**Review by agnivade — APPROVED** [review]

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

**Review by furqanmlk — APPROVED** [review]
This feature will be tested with 
https://github.com/mattermost/mattermost-server/pull/17375

**Comment by amyblais** [comment]
@wiggin77 Is this for the upcoming cloud release or a future release? I noticed it wasn't cherry-picked so I wanted to confirm.

**Comment by wiggin77** [comment]
> @wiggin77 Is this for the upcoming cloud release or a future release? I noticed it wasn't cherry-picked so I wanted to confirm.

Thanks @amyblais  My understanding is the window closed for the next cloud release.  I think it's fine to wait for the next one.  The feature is disabled by default.

**Pull request merged**